expand.py

```diff
@@ -678,6 +688,19 @@ def expand(cfg, lines, f, isfirst=False):
                     except:
                         cnt = "0"
                     f.write(cnt)
+                elif kw == "ASSIGN":
+                    args = argm.group(1).split(",")
+                    output = StringIO.StringIO()
+                    expand(cfg, [args[1]], output, isfirst)
+                    value = output.getvalue()
+                    output.close()
+                    try:
+                        v = cfg.eval_expr(value)
```

Comment:
Does this mean we're allowing arbitrary calculation expressions in the ASSIGN directive? If so, is cfg giving them access to some sort of shared namespace of values to pull from? This would be useful to document.

Reply:
Yes, the cfg is the current name space.  For the most part, this is just the defines in the .cfg file, but it is augmented with additional values when $$LOOPing.  (In particular, $$INDEX is the iteration counter, and if iterating over instantiations, all of the parameters are added to cfg.  We're doing a little dance to move INDEX to INDEX1, INDEX1 to INDEX2, and so on, whenever we start a new loop.)